controllers/workspace_controller_variable_sets.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	tfc "github.com/hashicorp/go-tfe"
+)
+
+func (w *workspaceInstance) getVariableSetIDs(ctx context.Context) ([]string, error) {
+	variableSets := w.instance.Spec.VariableSets
+
+	if variableSets == nil {
```

Comment:
This will always error, users also have the ability to not add variable sets
consider this....